Libraries/Text/TextInput/RCTBaseTextInputView.m

```diff
@@ -295,11 +295,20 @@ - (BOOL)textInputShouldChangeTextInRange:(NSRange)range replacementText:(NSStrin
   }
 
   if (_maxLength) {
-    NSUInteger allowedLength = _maxLength.integerValue - backedTextInputView.attributedText.string.length + range.length;
+    __block NSUInteger allowedLength = _maxLength.integerValue - backedTextInputView.attributedText.string.length + range.length;
 
     if (text.length > allowedLength) {
       // If we typed/pasted more than one character, limit the text inputted.
       if (text.length > 1) {
+        // Fix TextInput maxlength cuts Unicode caracters (Emojis) #10964
+        [text enumerateSubstringsInRange:NSMakeRange(0, text.length) options:NSStringEnumerationByComposedCharacterSequences usingBlock:^(NSString * _Nullable substring, NSRange substringRange, NSRange enclosingRange, BOOL * _Nonnull stop){
+            // ensure emoji character was not cut
+            if(substringRange.location + substring.length > allowedLength){
```

Comment:
@hzxiaosheng Hey, please rebase the branch first, sorry for that, it's our fault.
And I think we need treat any emoji as 1 glyph, so here may wrong, because we still use `substring`'s `length` to compare with `allowedLength`.